pkg/odo/cli/dev/dev.go

```diff
@@ -256,6 +267,14 @@ func regenerateComponentAdapterFromWatchParams(parameters watch.WatchParameters)
 	return adapters.NewComponentAdapter(parameters.ComponentName, parameters.Path, parameters.ApplicationName, devObj, platformContext)
 }
 
+func (o *DevOptions) HandleSignal() error {
+	o.cancel()
+	// At this point, `ctx.Done()` will be raised, and the cleanup will be done
+	// we are waiting for the cleanup to finish, before to continue the signal handler processing
+	for {
```

Comment:
Finally, I'm not sure it is a good practice to have a long running signal handler. Feel free to look at this patch for another handling solution: https://github.com/feloy/odo/commit/4b9b49029795704afec57186ec6b51ab64d39c4c

Reply:
Yeah, I am not sure either. The onus is on the handler to make sure that the program doesn't go into a state of waiting forever that would require something like a `SIGKILL`. That could be a good thing or a bad thing depending on the situation we are in. As far as `odo dev` in the current PR is concerned, it's handling it correctly, so it shouldn't be a problem.

How about we keep it this way for the time being and see if it becomes a problem? I could very well just modify the implementation to what you have in your commit, but if we are not sure that it's definitely bad practice, why not keep it instead...

Reply:
ok, let's keep it as this and see